core/src/network/peer.rs

```diff
@@ -432,7 +403,7 @@ where
     THandler: IntoConnectionHandler,
 {
     peer_id: PeerId,
-    network: &'a mut Network<TTrans, THandler>,
```

Comment:
Why this change? Are you planning on removing this field?

Reply:
Mostly for the sake of consistency. Each of the `XXXPeer`, e.g. `ConnectedPeer` and `DisconnectedPeer`, contain a `&mut Network`. Holding this mutable reference is relevant e.g. for `ConnectedPeer::some_connection`:

``` rust
    /// Obtains some established connection to the peer.
    pub fn some_connection(&mut self) -> EstablishedConnection<THandlerInEvent<THandler>> {
        self.connections()
            .into_first()
            .expect("By `Peer::new` and the definition of `ConnectedPeer`.")
    }
```

Given that `ConnectedPeer` is only created when the peer is actually connected and given that `ConnectedPeer` holds a `&mut Network` and thus a user could not disconnect the peer in the meantine, `ConnectedPeer::some_connection` can do the `expect` call.

Does that make sense @MarcoPolo?

> In the long run I would like to move and merge `libp2p_core::Network` and `libp2p_core::Pool` into  libp2p_swarm::Swarm` thus deduplicating `libp2p_core::DialOpts` and `libp2p_swarm::DialOpts`.

Following up on the above from the PR description, I would likely suggest getting rid of all of `peer.rs` in the same go. While I think the `Peer` construct offers a cool user experience, I don't think it makes sense for an rust-libp2p internal API.

Reply:
Yeah, I think that makes sense. Even though we never use `_network` anywhere, we still want to keep it so that we know that nothing else has a `&mut` on `Network`. Is that correct?